web3/_utils/async_transactions.py

```diff
@@ -115,7 +115,7 @@ async def fill_transaction_defaults(w3: "Web3", transaction: TxParams) -> TxPara
     strategy_based_gas_price = await w3.eth.generate_gas_price(  # type: ignore
         transaction
     )
-    is_dynamic_fee_transaction = not strategy_based_gas_price and (
```

Comment:
Eesh, good catch, thanks for fixing!

Reply:
No problem :) Also found the same bug in `TRANSACTION_DEFAULTS`